recipes/dbus/1.x.x/conanfile.py

```diff
@@ -54,69 +66,118 @@ def configure(self):
             if self.info.settings.compiler == "gcc" and Version(self.info.settings.compiler.version) < 7:
                 raise ConanInvalidConfiguration(f"{self.ref} requires at least gcc 7.")
-            if self.info.settings.os == "Windows":
-                raise ConanInvalidConfiguration(f"{self.ref} does not support windows. contributions are welcome")
+            
+        if not self._meson_available and self.info.settings.os == "Windows":
+                raise ConanInvalidConfiguration(f"{self.ref} does not support Windows. Contributions welcome.")
 
     def source(self):
         get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
 
     def generate(self):
-        tc = CMakeToolchain(self)
-        tc.variables["DBUS_BUILD_TESTS"] = False
-        tc.variables["DBUS_ENABLE_DOXYGEN_DOCS"] = False
-        tc.variables["DBUS_ENABLE_XML_DOCS"] = False
-        tc.variables["DBUS_BUILD_X11"] = self.options.get_safe("with_x11", False)
-        tc.variables["DBUS_WITH_GLIB"] = self.options.with_glib
-        tc.variables["DBUS_DISABLE_ASSERT"] = is_apple_os(self)
-        tc.variables["DBUS_DISABLE_CHECKS"] = False
-        # https://github.com/freedesktop/dbus/commit/e827309976cab94c806fda20013915f1db2d4f5a
-        tc.variables["DBUS_SESSION_SOCKET_DIR"] = self.options.session_socket_dir
-        tc.generate()
-
-        deps = CMakeDeps(self)
-        deps.generate()
+        if self._meson_available:
+            tc = MesonToolchain(self)
+            tc.project_options["asserts"] = not is_apple_os(self)
+            tc.project_options["checks"] = False
+            tc.project_options["doxygen_docs"] = "disabled"
+            tc.project_options["modular_tests"] = "disabled"
+            tc.project_options["session_socket_dir"] = str(self.options.session_socket_dir)
+            tc.project_options["selinux"] = "enabled" if self.options.get_safe("with_selinux", False) else "disabled"
+            tc.project_options["systemd"] = "enabled" if self.options.get_safe("with_systemd", False) else "disabled"
+            if self.options.get_safe("with_systemd", False):
+                tc.project_options["systemd_system_unitdir"] = os.path.join(self.package_folder, "lib", "systemd", "system")
+                tc.project_options["systemd_user_unitdir"] = os.path.join(self.package_folder, "lib", "systemd", "user")
+            if is_apple_os(self):
+                tc.project_options["launchd_agent_dir"] = os.path.join(self.package_folder, "res", "LaunchAgents")
+            tc.project_options["x11_autolaunch"] = "enabled" if self.options.get_safe("with_x11", False) else "disabled"
+            tc.project_options["xml_docs"] = "disabled"
+            tc.generate()
+            env = VirtualBuildEnv(self)
+            env.generate(scope="build")
+        else:
+            tc = CMakeToolchain(self)
+            tc.cache_variables["DBUS_BUILD_TESTS"] = False
```

Comment:
Any reason for changing from variables to cached_variables?

Reply:
I wonder if that got pulled in by accident from something I did previously. I do like using those CMake cache variables, though :) I've reverted this!

Reply:
Thanks!!